delphi-checks/src/main/java/au/com/integradev/delphi/checks/UnusedPropertyCheck.java

```diff
@@ -33,20 +34,35 @@
 public class UnusedPropertyCheck extends DelphiCheck {
   private static final String MESSAGE = "Remove this unused property.";
 
+  @RuleProperty(
+      key = "excludeApi",
+      description = "Exclude properties declared in the interface section with public visibility.")
+  public boolean excludeApi = false;
+
   @Override
   public DelphiCheckContext visit(PropertyNode property, DelphiCheckContext context) {
     NameDeclarationNode name = property.getPropertyName();
     PropertyNameDeclaration declaration = (PropertyNameDeclaration) name.getNameDeclaration();
-    if (isUnused(declaration)) {
+    if (isIssue(declaration)) {
       reportIssue(context, name, MESSAGE);
     }
     return context;
   }
 
-  private static boolean isUnused(PropertyNameDeclaration declaration) {
-    return !declaration.isPublished()
-        && declaration.getAttributeTypes().isEmpty()
-        && declaration.getScope().getOccurrencesFor(declaration).isEmpty()
-        && declaration.getRedeclarations().stream().allMatch(UnusedPropertyCheck::isUnused);
+  private boolean isIssue(PropertyNameDeclaration declaration) {
+    if (declaration.isPublished()) {
+      return false;
+    }
+
+    if (excludeApi && declaration.isPublic() && !declaration.isImplementationDeclaration()) {
```

Comment:
Shouldn't we also be excluding published types? e.g.

```delphi
interface

type
  TMyApiType = class(TObject)
  published type
    TMyNestedApiType = class(TObject)
    end;
  end;
```

Reply:
Yes, good call.

Reply:
sidenote: I think you meant for this comment to be in `UnusedTypeCheck`

Reply:
Yes I did, my bad.